src/states/common/approved.rs

```diff
@@ -137,8 +137,13 @@ pub trait Approved: Relocated {
                     // FIXME: Handle properly
                     unreachable!("...")
                 }
-                Observation::Genesis { .. } => {
+                Observation::Genesis {
+                    group,
+                    related_info,
+                } => {
                     // FIXME: Validate with Chain info.
```

Comment:
Is this `FIXME` still relevant?

Reply:
This is about validating the group value. Probably still relevant